### PR TITLE
Migrate TC tests/functional/snapshot/test_restore_online_vol.py

### PR DESCRIPTION
**Pull request by Sheetalpamecha**
[Migrate TC tests/functional/snapshot/test_restore_online_vol.py](https://github.com/gluster/glusto-tests/blob/master/tests/functional/snapshot/test_restore_online_vol.py)

Updates: #292 
Signed-off-by: Sheetal Pamecha <spamecha@redhat.com>

### CODE REVIEW

**Comment by nik-redhat** [comment]
Hey, @Sheetalpamecha can you please ass the results?

File: tests/functional/snapshot/test_restore_online_vol.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        3. Perform I/O on mounts
+        4. Create 1 snapshots snapy1
+        5. Validate snap created
+        6. Perform some more I/O
+        7. Create 1 more snapshot snapy2
+        8. Restore volume to snapy1
+          -- Restore should fail with message
+             "volume needs to be stopped before restore"
+        """
+        self.io_validation_complete = True
```
Comment (nik-redhat) [inline tests/functional/snapshot/test_restore_online_vol.py:58]: We don't seem to be checking the value of this variable anywhere, I think we can remove it.

File: tests/functional/snapshot/test_restore_online_vol.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        # Performing step 3 to 7 in loop here
+        for i in range(1, 3):
+            # Perform I/O
+            self.counter = 1
+            self.mounts = redant.es.get_mnt_pts_dict_in_list(self.vol_name)
```
Comment (nik-redhat) [inline tests/functional/snapshot/test_restore_online_vol.py:63]: We can move this definition out of the for loop, it is redefining the same value to an unchanged variable.

File: tests/functional/snapshot/test_restore_online_vol.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            if not redant.get_mounts_stat(self.mounts):
+                raise Exception("Stat on mountpoints failed.")
+
+            # Create snapshot
+            snapname = f"{self.vol_name}-snap{i}"
+            redant.snap_create(self.vol_name, snapname, self.server_list[0])
+
+            # Check for no of snaps using snap_list
+            snap_list = redant.get_snap_list(self.server_list[0])
+            if len(snap_list) != i:
```
Comment (nik-redhat) [inline tests/functional/snapshot/test_restore_online_vol.py:92]: Either check the 'count' -> `int(snap_list['msg']['snapList']['count'])` value or the 'snapshot' length using this way `len(snap_list['msg']['snapList']['snapshot'])`

**Review by nik-redhat — CHANGES_REQUESTED** [review]

**Comment by Sheetalpamecha** [comment]
> Hey, @Sheetalpamecha can you please add the results?

Hi Nikhil, I am facing some problem with my machines and i am not able to run snapshot TCs. Can you please take this PR forward?

**Comment by nik-redhat** [comment]
```
Results for snapshot
Test nature : disruptive
TestName : test_restore_online_vol
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:00:58       |      NA     |
|     dist    |     PASS    |        00:00:54       |      NA     |
|     disp    |     PASS    |        00:00:55       |      NA     |
|   dist-rep  |     PASS    |        00:01:10       |      NA     |
|  dist-disp  |     PASS    |        00:01:12       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   1   |
|         Non Disruptive Tests         |   0   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   5   |
|            Tests Skipped             |   0   |
|             Total Tests              |   1   |
|   Disruptive Tests Pass Percentage   | 100.0 |
| Non Disruptive Tests Pass Percentage |   0   |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 05 minutes 24 seconds
✔ Tearing down successful.
```

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**